wire-ios-sync-engine/Source/Synchronization/Strategies/UserClientEventConsumer.swift

```diff
@@ -25,38 +25,50 @@ import Foundation
 @objcMembers
 public class UserClientEventConsumer: NSObject, ZMEventAsyncConsumer {
 
-    let managedObjectContext: NSManagedObjectContext
-    let clientRegistrationStatus: ZMClientRegistrationStatus
-    let clientUpdateStatus: ClientUpdateStatus
+    private let managedObjectContext: NSManagedObjectContext
+    private let clientRegistrationStatus: ZMClientRegistrationStatus
+    private let clientUpdateStatus: ClientUpdateStatus
+    private let useCaseFactory: UseCaseFactoryProtocol
 
-    public init (managedObjectContext: NSManagedObjectContext,
-                 clientRegistrationStatus: ZMClientRegistrationStatus,
-                 clientUpdateStatus: ClientUpdateStatus) {
+    public init(
+        managedObjectContext: NSManagedObjectContext,
+        clientRegistrationStatus: ZMClientRegistrationStatus,
+        clientUpdateStatus: ClientUpdateStatus,
+        useCaseFactory: UseCaseFactoryProtocol
```

Comment:
Why do we need a use case factory? Why can't we inject `resolveOneOnOneUseCase` directly?

Reply:
yeah good point

Reply:
ah I think because I was thinking the use case should be created and invoked once, rather than injecting an instance to be invoked repeatedly. What's the best practice?

Reply:
Not completely sure about the best practice, but would agree with @typfel here. The use case should be a light-weight `struct` or `class` without states. So it should be save to retain it for long time also and pass around.

Additionally then I would question `UseCaseFactoryProtocol`, the factory can be used to create use cases with their protocols, but having another protocol here feels for me like a too complex construction.

Reply:
I'm on the same line as @markusfassbender,  since the use cases doesn't hold any state it fine to hold on to it an re-use it.

Reply:
Ok got it, thanks!

Reply:
done